File: MF1933118/swift/Calendar66.swift
/**类名_方法名(_形参类型)*
 *  @author: MF1933118 张锐汀
 *  @className: Calendar_dateComponents
 *  @apiSignature: swift.Calendar.dateComponents ( [('TimeZone', ['in', 'timeZone']), ('Date', ['from', 'date'])] )
 *  @description: Returns all the date components of a date.
 *  @Map: Calendar$func getInstance(TimeZone zone) -> time
 */


import Foundation
class Calendar_dateComponents{
    /**
     * input: 2
     * tz0 tz0=TimeZone.init(secondsFromGMT: 3600*7)
     * class0 class0=Calendar.current
     * output: 1
     * ret0  ret0=calendar.dateComponents(in: tz0, from: now)
     * Calendar$through-time$https://blog.csdn.net/shenjie_xsj/article/details/79033861
     */

    static func dateComponents0(){
        var class0=Calendar.current();
        var tz0=TimeZone.init(secondsFromGMT: 3600*7)
        var ret0=calendar.dateComponents(in: tz0, from: now);
        assert (tz0=TimeZone.init(secondsFromGMT: 3600*7));
        assert (ret0.calendar.dateComponents(in: TimeZone.init(secondsFromGMT: 3600*7), from: now));
        print(ret0);
    }

    /**
     * input: 2
     * tz0 tz0=TimeZone.init(secondsFromGMT: 3600*8)
     * class0 class0=Calendar.current
     * output: 1
     * ret0  ret0=calendar.dateComponents(in: TimeZone.init(secondsFromGMT: 3600*8)!, from: now)
     * Calendar$through-time$https://blog.csdn.net/shenjie_xsj/article/details/79033861
     */

    static func dateComponents1(){
        var class0=Calendar.current();
        var tz0=TimeZone.init(secondsFromGMT: 3600*8)
        var ret0=calendar.dateComponents(in: tz0, from: now);
        assert (tz0=TimeZone.init(secondsFromGMT: 3600*8));
        assert (ret0.calendar.dateComponents(in: TimeZone.init(secondsFromGMT: 3600*8), from: now));
        print(ret0);
    }
}
Calendar_dateComponents.dateComponents0()
Calendar_dateComponents.dateComponents1()

File: MF1933118/java/Calendar66.java
/**类名_方法名(_形参类型)*($返回值类型)*
 *  @author: MF1933118 张锐汀
 *  @className: Calendar_getInstance_TimeZone
 *  @apiSignature: java.util.calendar.getInstance(TimeZone zone)
 *  @description: Gets a calendar using the specified time zone and default locale
 *  @Map: Calendar$func DateComponents( [('TimeZone', ['in', 'timeZone']), ('Date', ['from', 'date'])])-> time
 */
public class Calendar_getInstance_TimeZone {
    /**
     * input: 2
     * tz0 tz0=TimeZone.getTimeZone("GMT")
     * class0 class0="GMT"
     * output: 1
     * ret0  ret0="GMT"
     */
    public  static void getInstance(TimeZone){
        System.out.println(">>>>>>>>>>>>");
        TimeZone tz0 = TimeZone.getTimeZone("GMT");
        Calendar class0 = Calendar.getInstance(tz0);  
        int ret0=class0.getInstance(class0);
        assert (tz0==TimeZone.getTimeZone("GMT"));
        assert (ret0.equals("GMT"));
        System.out.println(ret0);
    }

    /**
     * input: 2
     * tz1 tz1=TimeZone.getTimeZone("EST")
     * class0 class0="EST"
     * output: 1
     * ret0  ret0="EST"
     */
    public  static void getInstance(TimeZone){
        System.out.println(">>>>>>>>>>>>");
        TimeZone tz1 = TimeZone.getTimeZone("EST");
        Calendar class0 = Calendar.getInstance(tz1);  
        int ret0=class0.getInstance(class0);
        assert (tz1==TimeZone.getTimeZone("EST"));
        assert (ret0.equals("EST"));
        System.out.println(ret0);
    }


    public static void main(String[] args) {
        Calendar_getInstance_TimeZone.getInstance0(TimeZone);
        Calendar_getInstance_TimeZone.getInstance1(TimeZone);
    }
}